pkg/token/execCredentialPlugin.go

```diff
@@ -28,6 +32,15 @@ type execCredentialPlugin struct {
 }
 
 func New(o *Options) (ExecCredentialPlugin, error) {
+	env := os.Getenv(execInfoEnv)
+	//fmt.Fprintln(os.Stderr, os.Getenv(execInfoEnv))
+	var execCredential clientauthentication.ExecCredential
+	if error := json.Unmarshal([]byte(env), &execCredential); error != nil {
+		return nil, fmt.Errorf("cannot convert to ExecCredential: %w", error)
+	}
+	if !execCredential.Spec.Interactive && o.LoginMethod == DeviceCodeLogin {
```

Comment:
lets work on #140 first so that you can add appropriate unit test in this PR

Reply:
Added unit test for this commit 8523f02, please check it out. Thanks!